client/apps/edit/components/content2/sections/text/index.coffee

```diff
@@ -45,16 +47,23 @@ module.exports = React.createClass
 
   componentDidMount: ->
     if @props.section.get('body')?.length
-      blocksFromHTML = convertFromRichHtml @props.section.get('body')
-      editorState = EditorState.createWithContent(blocksFromHTML, new CompositeDecorator(decorators()))
+      html = standardizeSpacing @props.section.get('body')
+      unless @props.article.get('layout') is 'classic'
+        html = setContentStartEnd(html, @props.article.get('layout'), @props.isStartText, @props.isEndText)
+      blocksFromHTML = convertFromRichHtml html
+      editorState = EditorState.createWithContent(blocksFromHTML, new CompositeDecorator(decorators(@props.article.get('layout'))))
       editorState = setSelectionToStart(editorState) if @props.editing
       @setState
-        html: @props.section.get('body')
+        html: html
         editorState: editorState
     else if @props.editing
       @focus()
 
   componentDidUpdate: (prevProps) ->
+    if @props.isEndText != prevProps.isEndText or @props.isStartText != prevProps.isStartText
```

Comment:
These should probably use the coffeescript syntax:
`if @props.isEndText isnt prevProps.isEndText or @props.isStartText isnt prevProps.isStartText`

!= is a little risky to use, we could also do !== here instead which is what `isnt` evaluates to.